File: 35.search-position/sp.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number}
 */
var searchInsert = function(nums, target) {
    var lo = 0;
    var hi = nums.length - 1;
    while (lo <= hi) {
        var mid = parseInt((lo + hi) / 2, 10);
        var num = nums[mid];
        if (num < target) lo = mid + 1;
        else if (num > target) hi = mid - 1;
        else return mid;
    }
    return low;
};

File: 35.search-position/sp.cpp
#include <iostream>
#include <vector>
using namespace std;
class Solution {
public:
    int searchInsert (vector<int>& nums, int target) {
        int low = 0, high = nums.size() - 1;
        while (low <= high) {
            int mid = (low + high) / 2;
            int num = nums[mid];
            if (num < target) low = mid + 1;
            else high = mid - 1;
        }
        return low;
    }
};

int main (int argc, char const *argv[]) {
    vector<int> a(5, 3);
    Solution s;
    int res = s.searchInsert(a, 5);
    cout << res << endl;
    return 0;
}